src/newsfeed/infrastructure/subscription_storages.py

```diff
@@ -136,6 +142,129 @@ def __init__(self, config: Dict[str, str]):
+    async def get_by_to_newsfeed_id(self, newsfeed_id: str) -> Iterable[SubscriptionData]:
+        """Return subscriptions to specified newsfeed."""
+        async with self._get_connection() as redis:
+            subscribers_storage = []
+            for subscriber in await redis.lrange(
+                    key=f'subscribers:{newsfeed_id}',
+                    start=0,
+                    stop=-1,
+            ):
+                subscribers_storage.append(json.loads(subscriber))
+        return subscribers_storage
+
+    async def get_by_fqid(self, newsfeed_id: str, subscription_id: str) -> SubscriptionData:
+        """Return subscription of specified newsfeed."""
+        async with self._get_connection() as redis:
+            subscription = await redis.get(f'subscription:{subscription_id}')
+
+        if not subscription:
+            raise SubscriptionNotFound(
+                newsfeed_id=newsfeed_id,
+                subscription_id=subscription_id,
+            )
+        else:
+            return json.loads(subscription)
+
+    async def get_between(self, newsfeed_id: str, to_newsfeed_id: str) -> SubscriptionData:
+        """Return subscription between specified newsfeeds."""
+        async with self._get_connection() as redis:
+            subscription = await redis.get(
+                f"subscription_between:{newsfeed_id}{to_newsfeed_id}"
+            )
+
+        if not subscription:
+            raise SubscriptionBetweenNotFound(
+                newsfeed_id=newsfeed_id,
+                to_newsfeed_id=to_newsfeed_id,
+            )
+        else:
+            return json.loads(subscription)
+
+    async def add(self, subscription_data: SubscriptionData) -> None:
+        """Add subscription data to the storage."""
+        subscription_id = str(subscription_data['id'])
+        newsfeed_id = str(subscription_data['newsfeed_id'])
+        to_newsfeed_id = str(subscription_data['to_newsfeed_id'])
+
+        async with self._get_connection() as redis:
+            await redis.lpush(
+                key=f'subscriptions:{newsfeed_id}',
+                value=json.dumps(subscription_data),
+            )
+            await redis.lpush(
+                key=f'subscribers:{to_newsfeed_id}',
+                value=json.dumps(subscription_data),
+            )
+            await redis.append(
+                key=f'subscription:{subscription_id}',
+                value=json.dumps(subscription_data),
+            )
+            await redis.append(
+                key=f'subscription_between:{newsfeed_id}{to_newsfeed_id}',
+                value=json.dumps(subscription_data),
+            )
+
+    async def delete_by_fqid(self, newsfeed_id: str, subscription_id: str) -> None:
+        """Delete specified subscription."""
+        async with self._get_connection() as redis:
+            subscription_key = f'subscription:{subscription_id}'
+            subscription = await redis.get(subscription_key)
+            if not subscription:
```

Comment:
This `if / else` construction is a candidate for Guard Clause refactoring.

Links:

- https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html
- https://medium.com/better-programming/refactoring-guard-clauses-2ceeaa1a9da